providers/zone.rb

```diff
@@ -23,11 +23,34 @@ def whyrun_supported?
 
 use_inline_resources
 
+def load_current_resource
+  @current_resource = Chef::Resource::T3PdnsZone.new(new_resource.name)
+
+  @current_resource.name(new_resource.name)
+  @current_resource.exists = ::File.file?("/etc/powerdns/zones/#{new_resource.name}.zone")
+  @current_resource
+end
+
+# Returns the pdns_control reload command, depending on if
+# we create a new zone or update an existing one.
+def reload_command
+
```

Comment:
I fire a `/usr/bin/pdns_control reload` after zone changes, however a do not add new zones as most of my zones are coming from a mysql backend.